### PR TITLE
Dnspacket comboaddr

### PR DESCRIPTION
**Pull request by cmouse**

### CODE REVIEW

**Comment by cmouse** [comment]
Please review.

File: pdns/lua-auth.cc [hunk 1]
```diff
@@ -193,13 +193,13 @@ static int ldp_addRecords(lua_State *L) {
...
   DNSPacket *p=ldp_checkDNSPacket(L);
-  lua_pushstring(L, p->getRemote().c_str());
+  lua_pushstring(L, p->getRemote().toString().c_str());
   return 1;
 }
 
 static int ldp_getRemoteRaw(lua_State *L) {
   DNSPacket *p=ldp_checkDNSPacket(L);
-  const ComboAddress& ca=p->d_remote;
+  const ComboAddress& ca=p->getRemote();
```
Comment (Habbie) [inline pdns/lua-auth.cc]: why change this?

File: pdns/packethandler.cc [hunk 2]
```diff
@@ -763,8 +763,8 @@ int PacketHandler::trySuperMasterSynchronous(DNSPacket *p, const DNSName& tsigke
   try {
     Resolver resolver;
     uint32_t theirserial;
-    resolver.getSoaSerial(p->getRemote(),p->qdomain, &theirserial);    
-    resolver.resolve(p->getRemote(), p->qdomain, QType::NS, &nsset);
+    resolver.getSoaSerial(p->getRemote().toString(),p->qdomain, &theirserial);    
+    resolver.resolve(p->getRemote().toString(), p->qdomain, QType::NS, &nsset);
```
Comment (Habbie) [inline pdns/packethandler.cc]: Wonder if these should eventually be taking CA as well :)

**Comment by Habbie** [comment]
minor nits. LGTM.

File: pdns/lua-auth.cc (hunk 1 above)
Comment (cmouse) [inline pdns/lua-auth.cc]: it effectively returns same thing.

File: pdns/dnspacket.cc [hunk 3]
```diff
@@ -81,9 +81,9 @@ const string& DNSPacket::getString()
   return d_rawpacket;
 }
 
-string DNSPacket::getRemote() const
+ComboAddress DNSPacket::getRemote() const
```
Comment (rgacogne) [inline pdns/dnspacket.cc]: const ComboAddress& DNSPacket::getRemote() const, perhaps?

Comment (Habbie) [inline pdns/dnspacket.cc]: Won't & give us risk of leaking if we're not careful? CA is pretty small in any case.

File: pdns/packethandler.cc (hunk 2 above)
Comment (cmouse) [inline pdns/packethandler.cc]: Probably, but not in this change.

**Comment by cmouse** [comment]
Changed lua backend too.

**Comment by Habbie** [comment]
This looks good to me but there is no reason to merge it before 4.0.0, so I suggest doing it right after.

**Pull request merged**